[wasjoe1] ip

wasjoe1 opened the pull request:
### Jeoe chat bot
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
helps you to take down your daily tasks.

- text-based
- **easy** to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

download it from [here](https://github.com/wasjoe1/ip).
1. double-click it.
2. add your tasks.
3. let it manage your tasks for you 😉
4. And it is FREE!

Features:
- [ ] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Jeoe.instantiate();
        Jeoe.run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

wasjoe1 committed:
Level-0 commit by joe : 1st ever commit

wasjoe1 committed:
Level-1 commit by joe : just bye

wasjoe1 committed:
Level-1 commit by joe : just bye

wasjoe1 committed:
Level-2 commit by joe : add stuff & display a list

wasjoe1 committed:
Level-3 commit by joe : Having tasks

wasjoe1 committed:
Level-4 commit by joe : Have other task types

wasjoe1 committed:
Level-5 commit by joe : error handling

wasjoe1 committed:
A-TextUiTesting commit by joe : ui testing

wasjoe1 committed:
Level-6 commit by joe : deleting a task

wasjoe1 committed:
Add save feature to Jeoe.java and some restructure

wasjoe1 committed:
Teach chatbot to understand date & time in deadline task

wasjoe1 committed:
Add Manager and Command classes, more OOP

wasjoe1 committed:
Move all files and packages into Jeoe pacakage

wasjoe1 committed:
Merge branch 'add-gradle-support'

wasjoe1 committed:
Add gradle support to project

wasjoe1 committed:
Add 2 JUnit tests, Deadline & Todo

wasjoe1 committed:
Write javadocs for Jeoe program

wasjoe1 committed:
Follow the CS2103 coding standard for java

wasjoe1 committed:
Add find feature to Jeoe program

imkwokyong commented on src/main/java/Jeoe/Jeoe.java:
```diff
@@ -0,0 +1,89 @@
+package Jeoe;
```
Shouldn't the package names be all lowercase? I noticed this in other sub-packages too.

imkwokyong commented on src/main/java/Jeoe/Tasks/Task.java:
```diff
@@ -0,0 +1,118 @@
+     *
+     * @return Type of the task.
+     */
+    public TaskType taskType() {
```
Wouldn't it be better if the method name is `getTaskType` ?

imkwokyong commented on src/main/java/Jeoe/Tasks/TaskManager.java:
```diff
@@ -0,0 +1,152 @@
+            }
+            fileSc.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("storage manager did not load file properly");
```
Would it be better to delegate all printing tasks to the ui?

imkwokyong commented on src/main/java/Jeoe/Tasks/TaskManager.java:
```diff
@@ -0,0 +1,152 @@
+     *
+     * @param idxMark Index of the task to be marked.
+     */
+    public void mark(int idxMark) {
```
Wouldn't `markedIdx` be a better name for the parameter?

imkwokyong commented on src/main/java/Jeoe/Tasks/TaskManager.java:
```diff
@@ -0,0 +1,152 @@
+     *
+     * @param idxUnmark Index of the task to be un-marked.
+     */
+    public void unmark(int idxUnmark) {
```
Wouldn't `unmarkedIdx` be a better name for the parameter?

imkwokyong reviewed (commented):
I like how you commented on many details in your code, it makes the entire program much more readable!
Only some nits from my end - I believe the most important issue is the package naming. Otherwise, LGTM!

wasjoe1 committed:
Run checkstyle on code

sheryew commented on src/main/java/Jeoe/Commands/CommandParser.java:
I like the clean code (SWITCH) statements and how each command is successfully separated into their own class. This makes your code base very neat and easy to understand! 👍

sheryew commented on src/main/java/Jeoe/Tasks/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+    @Override
+    public String getEndDateTimeString() {
+        return this.by.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+        // return this.by.format(DateTimeFormatter.ofPattern("yyyy-MM-dd"))
```
Perhaps you can get rid of the commented code (Dead Code)?

sheryew commented:
Would honestly like to say job well done to you! Amazing work, very clean code with proper documentation and OOP principles. 👍 
However, I have added some comments that you can possible adopt.

wasjoe1 committed:
Add GUI for Jeoe program

wasjoe1 committed:
Include assertions in Main and Jeoe class

wasjoe1 committed:
Merge pull request #1 from wasjoe1/branch-A-Assertions

Include assertions in Main and Jeoe class

wasjoe1 committed:
Fill in update of Jeoe.java

wasjoe1 committed:
Update storage file

wasjoe1 committed:
Remove all style errors using checkstyle as a guide

wasjoe1 committed:
Merge pull request #2 from wasjoe1/branch-A-CodeQuality

Remove all style errors using checkstyle as a guide

wasjoe1 committed:
Clean code quality using checkstyle as a guide

Code quality is currently not clean. checkstyle shows that there
are errors in the styling of my code.

Additional space are being removed, as well as unnecessary code.
Checkstyle guide is used as reference.

These are done as then there would be a check list of styling
that i can check against for any errors.

wasjoe1 committed:
Merge branch 'master' into branch-A-Assertions

wasjoe1 committed:
Add yml file for workflow enhancement

Currently, every git action that is done, is done manually.

Add a file that has a yml extension for the workflow commands.

This is tedious and time consuming for a programmer. Some times
the programmer might even forget some of these commands.
Some of these commands can be automated with a workflow aid.

An additional file helps to abstract away the logic of the program and
program aid tools.

wasjoe1 committed:
Add sort feature to Jeoe

There is currently no sort feature.

Sometimes users want to see the list in a sorted manner for better
viewing.

Add a new sort command in Commands directory.
Add a sort method to Task manager.
Make Task objects comparable by extending comparable interface.
Add sort to the list of commands to the CommandParser.

Its done this way due to the abstraction of Jeoe.

wasjoe1 committed:
Merge pull request #3 from wasjoe1/branch-A-Assertions

Branch A-Assertions

wasjoe1 committed:
Upgrade GUI

Add stage title, configure the different components
to be dynamic when the user changes the window's screen size

wasjoe1 committed:
Add more error handling

wasjoe1 committed:
Edit UG

wasjoe1 committed:
edit the command summary

wasjoe1 committed:
Update Readme.md

wasjoe1 committed:
Update Readme.md

wasjoe1 committed:
Update README.md

wasjoe1 committed:
Add GFM configuration to render table

wasjoe1 committed:
Remove rendering of table

wasjoe1 committed:
Add more JUnit tests

wasjoe1 committed:
Add chat bot name to UG

wasjoe1 committed:
Rectify UG

wasjoe1 committed:
Change Jar file url
